aten/src/ATen/SparseCsrTensorUtils.h

```diff
@@ -369,5 +369,32 @@ inline bool only_sparse_compressed_add_trivial_cases(
       });
 }
 
+inline void only_sparse_binary_op_res_to_probably_int64_indices(
+    const Tensor& self,
+    const Tensor& other,
+    Tensor& out) {
+  const auto self_indices_dtype =
+      std::get<0>(at::sparse_csr::getCompressedPlainIndices(self))
+          .scalar_type();
+  const auto other_indices_dtype =
+      std::get<0>(at::sparse_csr::getCompressedPlainIndices(other))
+          .scalar_type();
+  const auto indices_common_dtype =
+      promoteTypes(self_indices_dtype, other_indices_dtype);
+  const auto out_indices_dtype =
+      std::get<0>(at::sparse_csr::getCompressedPlainIndices(out)).scalar_type();
+  if (out_indices_dtype != indices_common_dtype) {
+    Tensor compressed_indices, plain_indices;
+    std::tie(compressed_indices, plain_indices) =
+        at::sparse_csr::getCompressedPlainIndices(out);
```

Comment:
@cpuhrsch , maybe we should add a warning? Otherwise the docs mention that int64 indices get downcasted in the BLAS routines.